### PR TITLE
Update Plugins documentation section

### PR DESCRIPTION
**Pull request by bouthilx**
# Description

The doc was referring to the test algo gradient descent instead of
reusing the extensive documentation of the cookiecutter.

### CODE REVIEW

File: docs/src/plugins/algorithms.rst [hunk 1]
```diff
@@ -2,37 +2,246 @@
...
+
+Implementation
+==============
+
+.. code-block:: python
+
+    requires = 'real'
+
+Some algorithms requires the search space to be real. You can specify this requirement by adding the
+attribute ``requires = 'real'`` at the base of the class definition.In this case, the algorithm
```
Comment (Thomsch) [inline docs/src/plugins/algorithms.rst:136]:
```suggestion
attribute ``requires = 'real'`` at the base of the class definition. In this case, the algorithm
```

File: docs/src/plugins/algorithms.rst [hunk 2]
```diff
@@ -2,37 +2,246 @@
...
+wrapper in Orion's core will convert the search space to real one before passing it to your
+algorithm. This way the user can define discrete or categorital dimensions while using algorithms
```
Comment (Thomsch) [inline docs/src/plugins/algorithms.rst:138]:
```suggestion
algorithm. This way, the user can define discrete or categorital dimensions while using algorithms
```

**Comment by Thomsch** [comment]
In ´docs/src/plugins/install.rst´, we mention that
> Note that active development is currently focused on the core. Therefore only one algorithm plugin is available for now.

Is it still true?

**Comment by bouthilx** [comment]
Still true indeed. Hopefully it will soon not be true anymore. :)

**Comment by Thomsch** [comment]
Ok! I wasn't sure if soon and reality had collided yet :)

**Pull request merged**